### PR TITLE
[CALCITE-3716] ResultSetMetaData.getTableName should return empty str…

### PR DESCRIPTION
**Pull request by jinxing64**
…ing, not null, when column does not map to a table (Jin Xing)

### CODE REVIEW

**Comment by xy2953396112** [comment]
Looks good~ please fix ci

File: core/src/test/java/org/apache/calcite/test/TableInRootSchemaTest.java [hunk 1]
```diff
@@ -75,14 +74,14 @@
     final ResultSetMetaData resultSetMetaData = resultSet.getMetaData();
     assertThat(resultSetMetaData.getColumnName(1), equalTo("A"));
     assertThat(resultSetMetaData.getTableName(1), equalTo("SAMPLE"));
-    assertThat(resultSetMetaData.getSchemaName(1), nullValue());
+    assertThat(resultSetMetaData.getSchemaName(1), equalTo(""));
     assertThat(resultSetMetaData.getColumnClassName(1),
         equalTo("java.lang.String"));
     // Per JDBC, column name should be null. But DBUnit requires every column
     // to have a name, so the driver uses the label.
```
Comment (xy2953396112) [inline core/src/test/java/org/apache/calcite/test/TableInRootSchemaTest.java]: update doc ?

Comment (jinxing64) [reply core/src/test/java/org/apache/calcite/test/TableInRootSchemaTest.java]: The doc is for column name, which is not changed by this PR.

**Comment by jinxing64** [comment]
Thanks for approval @chunweilei

**Pull request closed without merging**